crates/commons/src/order.rs

```diff
@@ -7,6 +11,23 @@ use trade::ContractSymbol;
 use trade::Direction;
 use uuid::Uuid;
 
+#[derive(Serialize, Deserialize, Clone)]
+pub struct NewOrderRequest {
+    pub value: NewOrder,
+    /// A signature of the sha256 of [`value`]
+    pub signature: Signature,
+    /// The public key of the sending party
+    pub public_key: PublicKey,
+}
+
+impl NewOrderRequest {
+    pub fn verify(&self) -> Result<()> {
+        let message = self.value.message();
+        self.signature.verify(&message, &self.public_key)?;
```

Comment:
🤔 That doesn't make much sense does it?

Any body could send any message signed with the public key that they provide. We need to verify the signature on the node id of the user. This information can be found in the `trader_id` field.

Reply:
wups,that's the reason why we have PR reviews I guess 🙈